baggage.go

```diff
@@ -0,0 +1,113 @@
+package pyroscope
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/otel/baggage"
+)
+
+// FilterFunc returns true if this key should be used.
+type FilterFunc func(key string) bool
+
+// TransformFunc transforms the key.
+type TransformFunc func(key string) string
+
+// BaggageConfig contains configuration options for filtering and transforming
+// baggage members.
+type BaggageConfig struct {
+	filters    []FilterFunc
+	transforms []TransformFunc
+}
+
+// BaggageOption sets a option on a BaggageConfig.
+type BaggageOption func(config *BaggageConfig)
+
+// WithFilters sets filtering functions to apply to the baggage.
+func WithFilters(filters ...FilterFunc) BaggageOption {
+	return func(config *BaggageConfig) {
+		config.filters = append(config.filters, filters...)
+	}
+}
+
+// WithTransforms sets transformation functions to apply to the baggage.
+func WithTransforms(transforms ...TransformFunc) BaggageOption {
+	return func(config *BaggageConfig) {
+		config.transforms = append(config.transforms, transforms...)
+	}
+}
+
+// K6Options provides default options to select k6 members from the baggage.
+func K6Options() []BaggageOption {
+	return []BaggageOption{
+		WithFilters(func(key string) bool {
+			return strings.HasPrefix(key, "k6.")
+		}),
+		WithTransforms(func(key string) string {
+			return strings.ReplaceAll(key, ".", "_")
+		}),
+	}
+}
```

Comment:
Here's the code that sets the baggage members: https://github.com/grafana/jslib.k6.io/blob/346206a8d62c99263adf4a7437057820262d441b/lib/http-instrumentation-pyroscope/1.0.0/index.js#L10-L12.

As it stands right now, k6 will send us the follow labels:

- `k6.test_run_id`
- `k6.scenario`
- `k6.name` (AKA "group name")

A rough upper bound estimate suggests these unique label combinations would be measured in thousands or less. A single test will produce 1 `k6.test_run_id`, which generally will have no more than 30 `k6.scenario`, and each scenario would likely have no more than 50 `k6.name`. It's unlikely a test is run more than 10 times an hour.

That means each test run will generate 10 • 30 • 50 = **1500 unique label sets per hour**. Keep in mind this is being very generous, the vast majority of k6 tests will have significantly fewer scenarios, groups, and test runs per hour. Realistically, a test will be ran less than once an hour, have fewer than 5 scenarios, and fewer than 5 groups. A more "typical" estimate would be 1 • 5 • 5 = **25 unique label sets per hour**.

The biggest risk here is `k6.name`. If the group name is not specified, it will use the url of the request, which could easily grow unbounded. k6 will stop the test if there's more than 100k unique group names. If we can't accept this risk, we can filter the group names here, or push a change to the upstream k6 lib and make the change there.

Reply:
Totally makes sense! Thank you for the clarification!